modules/swagger-codegen/src/main/resources/typescript-angular2/api.mustache

```diff
@@ -18,10 +18,10 @@ export class {{classname}} {
     protected basePath = '{{basePath}}';
     public defaultHeaders : Headers = new Headers();
 
-    constructor(protected http: Http, basePath: string) {
```

Comment:
Shall we keep the constructor with the basePath argument? For other clients, we've similar constructor to set a different base path so that developers/owners can easily switch to a different environment (e.g sandbox, development) with a different base path (e.g. https://sandbox.api.testing.com)

you're welcome to add a new one for `constructor(protected http: Http)